appinventor/blocklyeditor/src/blocks/colors.js

```diff
@@ -11,178 +11,36 @@
-    this.appendDummyInput().appendField(new Blockly.FieldColour('#cccccc'), 'COLOR');
-    this.setOutput(true, AI.BlockUtils.YailTypeToBlocklyType("number",AI.BlockUtils.OUTPUT));
-    this.setTooltip(Blockly.Msg.LANG_COLOUR_MAKE_COLOUR_TOOLTIP);
-  },
-  typeblock: [{ translatedName: Blockly.Msg.LANG_COLOUR_LIGHT_GRAY }]
-};
-
-Blockly.Blocks['color_gray'] = {
-  // Colour picker.
-  category: "Colors",
-  helpUrl: Blockly.Msg.LANG_COLOUR_PICKER_HELPURL,
-  init: function() {
-    this.setColour(Blockly.COLOR_CATEGORY_HUE);
-    this.appendDummyInput().appendField(new Blockly.FieldColour('#888888'), 'COLOR');
-    this.setOutput(true, AI.BlockUtils.YailTypeToBlocklyType("number",AI.BlockUtils.OUTPUT));
-    this.setTooltip(Blockly.Msg.LANG_COLOUR_MAKE_COLOUR_TOOLTIP);
-  },
-  typeblock: [{ translatedName: Blockly.Msg.LANG_COLOUR_GRAY }]
-};
-
-
-Blockly.Blocks['color_dark_gray'] = {
-  // Colour picker.
-  category: "Colors",
-  helpUrl: Blockly.Msg.LANG_COLOUR_PICKER_HELPURL,
-  init: function() {
-    this.setColour(Blockly.COLOR_CATEGORY_HUE);
-    this.appendDummyInput().appendField(new Blockly.FieldColour('#444444'), 'COLOR');
-    this.setOutput(true, AI.BlockUtils.YailTypeToBlocklyType("number",AI.BlockUtils.OUTPUT));
-    this.setTooltip(Blockly.Msg.LANG_COLOUR_MAKE_COLOUR_TOOLTIP);
-  },
-  typeblock: [{ translatedName: Blockly.Msg.LANG_COLOUR_DARK_GRAY }]
-};
+createColorBlock('color_black', '#000000ff', Blockly.Msg.LANG_COLOUR_BLACK);
+createColorBlock('color_white', '#ffffffff', Blockly.Msg.LANG_COLOUR_WHITE);
+createColorBlock('color_red', '#ff0000ff', Blockly.Msg.LANG_COLOUR_RED);
+createColorBlock('color_pink', '#ffafafff', Blockly.Msg.LANG_COLOUR_PINK);
+createColorBlock('color_orange', '#ffc800ff', Blockly.Msg.LANG_COLOUR_ORANGE);
+createColorBlock('color_yellow', '#ffff00ff', Blockly.Msg.LANG_COLOUR_YELLOW);
+createColorBlock('color_green', '#00ff00ff', Blockly.Msg.LANG_COLOUR_GREEN);
+createColorBlock('color_cyan', '#00ffffff', Blockly.Msg.LANG_COLOUR_CYAN);
+createColorBlock('color_blue', '#0000ffff', Blockly.Msg.LANG_COLOUR_BLUE);
+createColorBlock('color_magenta', '#ff00ffff', Blockly.Msg.LANG_COLOUR_MAGENTA);
+createColorBlock('color_light_gray', '#ccccccff', Blockly.Msg.LANG_COLOUR_LIGHT_GRAY);
+createColorBlock('color_gray', '#888888ff', Blockly.Msg.LANG_COLOUR_GRAY);
+createColorBlock('color_dark_gray', '#444444ff', Blockly.Msg.LANG_COLOUR_DARK_GRAY);
+
+function createColorBlock(name, color, lang) {
+  Blockly.Blocks[name] = {
+    category: "Colors",
+    helpUrl: Blockly.Msg.LANG_COLOUR_PICKER_HELPURL,
+    init: function() {
+      this.setColour(Blockly.COLOR_CATEGORY_HUE);
+      this.appendDummyInput().appendField(new Blockly.FieldColor(color), 'COLOR');
+      this.setOutput(true, AI.BlockUtils.YailTypeToBlocklyType("number", AI.BlockUtils.OUTPUT));
+      this.setTooltip(Blockly.Msg.LANG_COLOUR_PICKER_TOOLTIP);
+    },
+    typeblock: [{ translatedName: lang }]
+  };
+}
```

Comment:
Stylistically, I'd prefer the function definition above it's usage starting on line 17.